arch/x86_64/inject-content.S

```diff
@@ -0,0 +1,82 @@
+#include "utils/asm.h"
+
+GLOBAL(inject_content)
+	/*
+	 * rsi is going to contain the address of free().
+	 * it's going to get wiped out by the call to malloc(),
+	 * so save it on the stack for later
+	 */
+	push %rsi
+	/*
+	 * same thing for rdx, which will contain the address
+	 * of _dl_open()
+	 */
+	push %rdx
+
+	/*
+	 * save previous value of r9,
+	 * because we're going to use it to call malloc()
+	 */
+	push %r9
+	// now move the address of malloc() into r9
+	mov %rdi,%r9
+	/*
+	 * choose the amount of memory to allocate with malloc()
+	 * based on the size of the path to the shared library
+	 * passed via rcx
+	 */
+	mov %rcx,%rdi
+	// now call r9; malloc()
+	callq *%r9
+	// after returning from malloc(),
+	// pop the previous value of r9 off the stack
+	pop %r9
+	// break in so that we can see what malloc() returned
+	int $3
+	// get the address of dlopen() off of
+	// the stack so we can call it
+	pop %rdx
+	// as before, save the previous value of r9 on the stack
+	push %r9
+	// copy the address of dlopen() into r9
+	mov %rdx,%r9
+	// 1st argument to dlopen():
+	// filename = the address of the buffer returned by malloc()
+	mov %rax,%rdi
+	// 2nd argument to dlopen(): flag = RTLD_LAZY
+	movabs $1,%rsi
+	// call dlopen()
+	callq *%r9
+	// restore old r9 value
+	pop %r9
+	// break in so that we can see what dlopen() returned
+	int $3
```

Comment:
Ditto.

Reply:
Ditto means he wants to repeat the same comment for this. Please see above. He's saying as follows: 
> "Please add a blank line and some more comments so that it can be noticed it's executed in a separate phase."

Reply:
And I mentioned about "//" because it is not used in Linux kernel coding style.